src/ctx.ts

```diff
@@ -0,0 +1,20 @@
+import * as vscode from 'vscode';
+import { SPMPackage } from './package';
+
+export class Ctx {
+	private constructor(
+        public workspaceRoot: string,
+        public extContext: vscode.ExtensionContext,
+        public spmPackage: SPMPackage
+    ) {}
+
+    static async create(
```

Comment:
What's the motivation for using a factory method here?

I understand the general benefits of factory methods, but if there's no concrete need for one here, I'd prefer not to include them. Every layer of abstraction we add just makes the codebase harder to get into for newcomers.

Reply:
It is an async function. You cannot have async constructors

Reply:
Thanks, I didn't know that yet!